datanode/dehistory/store/store.go

```diff
@@ -521,6 +521,19 @@ func (p *Store) extractHistorySegmentToStagingArea(ctx context.Context, toHeight
 	return nil
 }
 
+func (p *Store) HasHistorySegmentForHeight(toHeight int64) (bool, error) {
+	_, err := p.index.Get(toHeight)
```

Comment:
Could we invert the logic and not have so much indentation?
```
func (p *Store) HasHistorySegmentForHeight() {
    _, err := p.index.Get(toHeight)
    if err == nil {
        return true, nil
    }

    if errors.Is(err, ErrIndexEntryNotFound) {
        return false, nil
    }

    return false, fmt.Errorf("failed to get index entry for height:%d:%w", toHeight, err)
}
```

Reply:
code removed - no longer needed